src/Payments/Integrations/Cheque.php

```diff
@@ -13,56 +13,65 @@
 use Exception;
 use WC_Stripe_Payment_Request;
 use WC_Stripe_Helper;
+use Automattic\WooCommerce\Blocks\Assets\Api;
 
 /**
  * Cheque payment method integration
  *
  * @since $VID:$
  */
-final class Cheque extends AbstractPaymentMethodIntegration {
+final class Cheque extends AbstractPaymentMethodType {
 	/**
-	 * Payment method name.
+	 * Settings from the WP options table
 	 *
-	 * @var string
+	 * @var array
 	 */
-	protected $payment_method_name = 'cheque';
+	private $cheque_settings;
 
 	/**
-	 * Settings from the WP options table
+	 * An instance of the Asset Api
 	 *
-	 * @var array
+	 * @var Api
 	 */
-	private $cheque_settings;
+	private $asset_api;
 
 	/**
-	 * Registers the block type with WordPress.
+	 * Constructor
+	 *
+	 * @param string $name The name of the payment method type.
+	 * @param Api    $asset_api An instance of Api.
 	 */
-	public function register_payment_method() {
+	public function __construct( $name, Api $asset_api ) {
+		$this->asset_api = $asset_api;
+		parent::__construct( $name );
+	}
+
+	/**
+	 * Initializes the payment method type.
+	 */
+	public function initialize() {
```

Comment:
We still delay initialization and hitting options until needed. This probably could even be delayed until sometime in `PaymentAssets` (instead of `Library`)?